In [8]:
%matplotlib inline
import pandas as pd

In [14]:
titles = pd.read_csv('titles.csv')
titles.head()

title  year
0    Tasveer Mere Sanam  1996
1              Only You  1994
2  El pueblo del terror  1970
3               Machine  2007
4                  MARy  2008

### How many movies are listed in the titles dataframe?

In [27]:
titles.count()
len(titles.title)

244914

There are 244914 movies listed in titles dataframe

### What are the earliest two films listed in the titles dataframe?

In [25]:
earliest=titles.sort_values('year')
earliest.head(2)

title  year
193222          Miss Jerry  1894
51388   The Startled Lover  1898

Miss Jerry & The Startled Lover are the two earliest movies listed in titles dataframe

### How many movies have the title "Hamlet"?

In [40]:
hamlet=titles[titles['title']=="Hamlet"]
print(hamlet)
len(hamlet)

         title  year
6422    Hamlet  2017
10637   Hamlet  1921
39916   Hamlet  1913
52521   Hamlet  1910
62000   Hamlet  2000
67406   Hamlet  1969
82130   Hamlet  1987
88065   Hamlet  2016
101728  Hamlet  1948
122827  Hamlet  1973
133921  Hamlet  1976
154029  Hamlet  2015
161324  Hamlet  1964
172640  Hamlet  1954
173033  Hamlet  2011
178222  Hamlet  1996
189093  Hamlet  2009
192142  Hamlet  1990
198234  Hamlet  1911
202025  Hamlet  2014


20

### How many movies are titled "North by Northwest"?

In [41]:
NorthbyNorthwest=titles[titles['title']=="North by Northwest"]
print(NorthbyNorthwest)
len(NorthbyNorthwest)

                    title  year
74398  North by Northwest  1959


1

### When was the first movie titled "Hamlet" made?

In [55]:
earliest1=titles.sort_values('year')
earliest1[titles['title']=='Hamlet'].head(1)

C:\Users\ykbha\anaconda3\lib\site-packages\ipykernel_launcher.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  


title  year
52521  Hamlet  1910

### List all of the "Treasure Island" movies from earliest to most recent.

In [56]:
earliest1[titles['title']=='Treasure Island']

C:\Users\ykbha\anaconda3\lib\site-packages\ipykernel_launcher.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  """Entry point for launching an IPython kernel.


title  year
55686   Treasure Island  1918
196514  Treasure Island  1920
141822  Treasure Island  1934
77354   Treasure Island  1950
65051   Treasure Island  1972
37238   Treasure Island  1973
173051  Treasure Island  1985
27076   Treasure Island  1999

### How many movies were made in the year 1950?

In [71]:
len(titles[titles['year']>1950])

208820

### How many movies were made in the year 1960?

In [70]:
len(titles[titles['year']>1960])

195450

### How many movies were made from 1950 through 1959?

In [67]:
len(titles[(titles['year']>1950) & (titles['year']<1959)])

10306

### In what years has a movie titled "Batman" been released?

In [75]:
batman=earliest1[titles['title']=='Batman']
batman['year']

C:\Users\ykbha\anaconda3\lib\site-packages\ipykernel_launcher.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  """Entry point for launching an IPython kernel.


39349     1943
240896    1989
Name: year, dtype: int64

### How many roles were there in the movie "Inception"?

### How many roles in the movie "Inception" are NOT ranked by an "n" value?

### But how many roles in the movie "Inception" did receive an "n" value?

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

### How many roles were credited in the silent 1921 version of Hamlet?

### How many roles were credited in Branagh’s 1996 Hamlet?

### How many "Hamlet" roles have been listed in all film credits through history?

### How many people have played an "Ophelia"?

### How many people have played a role called "The Dude"?

### How many people have played a role called "The Stranger"?

### How many roles has Sidney Poitier played throughout his career?

### How many roles has Judi Dench played?

### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

### List the leading roles that Cary Grant played in the 1940s in order by year.

### How many roles were available for actors in the 1950s?

### How many roles were available for actresses in the 1950s?

### How many leading roles (n=1) were available from the beginning of film history through 1980?

### How many non-leading roles were available through from the beginning of film history through 1980?

### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?